### PR TITLE
Add profiler instructions when labeling issue with cpu

### PR DESCRIPTION
**Pull request by joostlek**
Be able to add debug information to certain types of labels. This would help triaging, as people who do triaging to give debug information without knowing that integration or feature themselves.

The goal for this PR is making able to give profiler debug information with only labeling the issue `cpu`. If someone who knows how the profiler works comes along, the user already knows what they should be doing and hopefully has it delivered already.

### CODE REVIEW

File: services/bots/src/github-webhook/handlers/debug_info.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const COMMENTS = {
+    'cpu': ' - Start a new issue, choose profiler as the integration\n - Include a callgrind file from the <https://www.home-assistant.io/integrations/profiler/> `profiler.start` service\n - Include a `py-spy` from <https://community.home-assistant.io/t/instructions-to-install-py-spy-on-haos/480473>. (be sure to zip them up before posting as GitHub mutates svg files)\n - If memory is leaking, include 4 hours of RAW logs (`Settings -> System -> Logs -> Download`) from the <https://www.home-assistant.io/integrations/profiler/> `profiler.start_log_objects` service with the default settings'
+}
+
+export class DebugInfo extends BaseWebhookHandler {
+    public allowedEventTypes = [EventType.ISSUES_LABELED];
+    public allowedRepositories = [HomeAssistantRepository.CORE];
+
+    async handle(context: WebhookContext<IssuesLabeledEvent>) {
+        if (!context.payload.label && Object.values(COMMENTS).indexOf(context.payload.label.toString()) > -1) {
```
Comment (ludeeus) [inline services/bots/src/github-webhook/handlers/debug_info.ts:15]: For the enabled event, `label` will always be in the payload.

File: services/bots/src/github-webhook/handlers/debug_info.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { IssuesLabeledEvent } from '@octokit/webhooks-types';
+import { EventType, HomeAssistantRepository } from '../github-webhook.const';
+import { WebhookContext } from '../github-webhook.model';
+import { BaseWebhookHandler } from './base';
+
+const COMMENTS = {
+    'cpu': ' - Start a new issue, choose profiler as the integration\n - Include a callgrind file from the <https://www.home-assistant.io/integrations/profiler/> `profiler.start` service\n - Include a `py-spy` from <https://community.home-assistant.io/t/instructions-to-install-py-spy-on-haos/480473>. (be sure to zip them up before posting as GitHub mutates svg files)\n - If memory is leaking, include 4 hours of RAW logs (`Settings -> System -> Logs -> Download`) from the <https://www.home-assistant.io/integrations/profiler/> `profiler.start_log_objects` service with the default settings'
+}
+
+export class DebugInfo extends BaseWebhookHandler {
```
Comment (ludeeus) [inline services/bots/src/github-webhook/handlers/debug_info.ts:10]: `LabelComment`  is probably a better name here.

File: services/bots/src/github-webhook/handlers/debug_info.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    public allowedEventTypes = [EventType.ISSUES_LABELED];
+    public allowedRepositories = [HomeAssistantRepository.CORE];
+
+    async handle(context: WebhookContext<IssuesLabeledEvent>) {
+        if (!context.payload.label && Object.values(COMMENTS).indexOf(context.payload.label.toString()) > -1) {
+            return;
+        }
+        context.scheduleIssueComment({
+            handler: 'DebugInfo',
+            comment: COMMENTS[context.payload.label],
```
Comment (ludeeus) [inline services/bots/src/github-webhook/handlers/debug_info.ts:20]: `context.payload.label` is an object, `context.payload.label.name` needs to be used.
<https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=labeled#issues>

File: services/bots/src/github-webhook/handlers/debug_info.ts (hunk 1 above)
Comment (ludeeus) [reply services/bots/src/github-webhook/handlers/debug_info.ts:15]:
```suggestion
        const comment = COMMENTS[context.payload.label.name]
        if (!comment) {
```

File: services/bots/src/github-webhook/handlers/debug_info.ts (hunk 3 above)
Comment (ludeeus) [reply services/bots/src/github-webhook/handlers/debug_info.ts:20]:
```suggestion
            comment,
```

File: services/bots/src/github-webhook/handlers/debug_info.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { IssuesLabeledEvent } from '@octokit/webhooks-types';
+import { EventType, HomeAssistantRepository } from '../github-webhook.const';
+import { WebhookContext } from '../github-webhook.model';
+import { BaseWebhookHandler } from './base';
+
+const COMMENTS = {
+    'cpu': ' - Start a new issue, choose profiler as the integration\n - Include a callgrind file from the <https://www.home-assistant.io/integrations/profiler/> `profiler.start` service\n - Include a `py-spy` from <https://community.home-assistant.io/t/instructions-to-install-py-spy-on-haos/480473>. (be sure to zip them up before posting as GitHub mutates svg files)\n - If memory is leaking, include 4 hours of RAW logs (`Settings -> System -> Logs -> Download`) from the <https://www.home-assistant.io/integrations/profiler/> `profiler.start_log_objects` service with the default settings'
```
Comment (ludeeus) [inline services/bots/src/github-webhook/handlers/debug_info.ts:7]: Use [my](https://my.home-assistant.io/create-link/) links when you tell the user to navigate inside Home Assistant.

Comment (ludeeus) [reply services/bots/src/github-webhook/handlers/debug_info.ts:7]: > Start a new issue, choose profiler as the integration
That part does not make sense to me, they have an issue already.

If they really need to create a separate one, give them a link that does that and add the correct integration for them.

Like <https://github.com/home-assistant/core/issues/new?assignees=&labels=&projects=&template=bug_report.yml&integration_name=profiler&integration_link=https%3A%2F%2Fwww.home-assistant.io%2Fintegrations%2Fprofiler>

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: services/bots/src/github-webhook/handlers/debug_info.ts (hunk 4 above)
Comment (joostlek) [reply services/bots/src/github-webhook/handlers/debug_info.ts]: Yea this is what @bdraco came up with. Can you maybe elaborate on the new issue part? Most of these issues are created without any integration link, so we should be able to add profiler label ourselves and add the cpu label to add these instructions

**Pull request closed without merging**